src/routes/v1/oauth/callback/post.js

```diff
@@ -41,15 +44,24 @@ export default async function(req, res, next) {
     ? tokenSet.claims.email.toLowerCase()
     : tokenSet.claims.preferred_username
     ? tokenSet.claims.preferred_username.toLowerCase()
+    : get(tokenSet.claims, claimsFallback.email)
+    ? get(tokenSet.claims, claimsFallback.email).toLowerCase()
```

Comment:
@ianstanton we should probably break this triple ternary up so its easier to reason about.

Reply:
```javascript
function getFirstMatchingClaim(claims, ...names) {
  for (let claim of names) {
    const val = get(claims, name);
    if (val !== undefined) return val;
  }
}
```

And then we dan do this as

```suggestion
    const email = _.lowerCase(getFirstMatchingClaim(tokenSet.claims, 'email', 'preferred_username', ...claimsFallback.email))
```

This also supports a list of fallback claims.